source/slang/slang-emit-cpp.cpp

```diff
@@ -2095,21 +2093,39 @@ void CPPSourceEmitter::emitIntrinsicCallExprImpl(
         else
         {
             // The user is invoking a built-in subscript operator
-            auto prec = getInfo(EmitOp::Postfix);
-            needClose = maybeEmitParens(outerPrec, prec);
 
-            emitOperand(args[0].get(), leftSide(outerPrec, prec));
-            m_writer->emit("[");
-            emitOperand(args[1].get(), getInfo(EmitOp::General));
-            m_writer->emit("]");
+            // Determine if we are calling the `ref` accessor:
+            // `ref` accessor returns a pointer of element type.
```

Comment:
This isn't a problem with this PR, but the steps required to special-case subscript calls here are out of control, and we really need to go back and triage the Slang stdlib so that all the built-in subscripts and their accessors have their emit logic defined in the stdlib instead of here.